src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -108,37 +138,24 @@ internal string GetExecutionPolicy(ConfigScope scope, string shellId)
 
         internal void RemoveExecutionPolicy(ConfigScope scope, string shellId)
         {
-            string scopeDirectory = psHomeConfigDirectory;
-
-            // Defaults to system wide.
-            if (ConfigScope.CurrentUser == scope)
-            {
-                scopeDirectory = appDataConfigDirectory;
-            }
-
-            string fileName = Path.Combine(scopeDirectory, configFileName);
             string valueName = string.Concat(shellId, ":", "ExecutionPolicy");
-            RemoveValueFromFile<string>(fileName, valueName);
+            RemoveValueFromFile<string>(scope, valueName);
         }
 
         internal void SetExecutionPolicy(ConfigScope scope, string shellId, string executionPolicy)
         {
-            string scopeDirectory = psHomeConfigDirectory;
-
             // Defaults to system wide.
             if (ConfigScope.CurrentUser == scope)
             {
                 // Exceptions are not caught so that they will propagate to the
                 // host for display to the user.
                 // CreateDirectory will succeed if the directory already exists
                 // so there is no reason to check Directory.Exists().
-                Directory.CreateDirectory(appDataConfigDirectory);
-                scopeDirectory = appDataConfigDirectory;
+                Directory.CreateDirectory(perUserConfigFile.Directory.FullName);
```

Comment:
I don't think it's necessary to use `FileInfo` type for `systemWideConfigFile` and `perUserConfigFile` to just use the `Directory` property here. Seems too heavy to me. I suggest we just add a private field `perUserConfigDirectory` to hold the directory path, and make `systemWideConfigFile` and `perUserConfigFile` strings.